pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,249 @@
+}
+
+// syncToStdout is the business logic of the controller. In this controller it simply prints
+// information about the deployment to stdout. In case an error happened, it has to simply return the error.
+// The retry logic should not be part of the business logic.
+func (c *StashController) runBackupConfigurationInjector(key string) error {
+	obj, exists, err := c.bcInformer.GetIndexer().GetByKey(key)
+	if err != nil {
+		glog.Errorf("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+	if !exists {
+		glog.Errorf("BackupConfiguration %s does not exit anymore\n", key)
+
+	} else {
+		backupconfiguration := obj.(*api_v1beta1.BackupConfiguration)
+		glog.Infof("Sync/Add/Update for BackupConfiguration %s", backupconfiguration.GetName())
+		if backupconfiguration.DeletionTimestamp != nil {
+			if core_util.HasFinalizer(backupconfiguration.ObjectMeta, api_v1beta1.StashKey) {
+				namespace, name, err := cache.SplitMetaNamespaceKey(key)
+				if err != nil {
+					return err
+				}
+				if err = c.EnsureV1beta1SidecarDeleted(backupconfiguration); err != nil {
+					return err
+				}
+				if err = c.EnsureCronJobDeleted(namespace, name); err != nil {
+					return err
+				}
+				// Remove finalizer
+				_, _, err = util2.PatchBackupConfiguration(c.stashClient.StashV1beta1(), backupconfiguration, func(in *api_v1beta1.BackupConfiguration) *api_v1beta1.BackupConfiguration {
+					in.ObjectMeta = core_util.RemoveFinalizer(in.ObjectMeta, api_v1beta1.StashKey)
+					return in
+
+				})
+				if err != nil {
+					return err
+				}
+			}
+		} else {
+			//Add finalizer
+			_, _, err := util2.PatchBackupConfiguration(c.stashClient.StashV1beta1(), backupconfiguration, func(in *api_v1beta1.BackupConfiguration) *api_v1beta1.BackupConfiguration {
+				in.ObjectMeta = core_util.AddFinalizer(in.ObjectMeta, api_v1beta1.StashKey)
+
+				return in
+			})
+			if err != nil {
+				return err
+			}
+			if backupconfiguration.Spec.Target != nil {
+				if backupconfiguration.Spec.Target.Ref != nil {
+					kind := backupconfiguration.Spec.Target.Ref.Kind
+					if util.BackupModel(kind) == util.ModelSidecar {
+						if err := c.EnsureV1beta1Sidecar(backupconfiguration); err != nil {
+							return err
+						}
+					}
+				}
+				if err := c.EnsureCronJob(backupconfiguration); err != nil {
+					return err
+				}
+			}
+		}
+
+	}
+	return nil
+}
+
+func (c *StashController) EnsureV1beta1SidecarDeleted(backupconfiguration *api_v1beta1.BackupConfiguration) error {
+	if backupconfiguration == nil {
+		return fmt.Errorf("BackupConfiguration is nil")
+	}
+	if backupconfiguration.Spec.Target != nil {
+		if backupconfiguration.Spec.Target.Ref != nil {
+			kind := backupconfiguration.Spec.Target.Ref.Kind
+			namespace := backupconfiguration.Namespace
+			resource_name := backupconfiguration.Spec.Target.Ref.Name
+			if err := c.CreateOrDeleteEvent(kind, namespace, resource_name); err != nil {
```

Comment:
Rename `c.CreateOrDeleteEvent()` to `c.sendEventToWorkloadQueue()`

Reply:
fixed